hrtime.h

```diff
@@ -91,6 +92,22 @@ rb_hrtime_add(rb_hrtime_t a, rb_hrtime_t b)
     return c;
 }
 
+static inline rb_hrtime_t
+rb_hrtime_sub(rb_hrtime_t a, rb_hrtime_t b)
```

Comment:
I need `rb_hrtime_sub` to be able to compute the relative time from the absolute one. Since the sleep may be interrupted for other reasons and resumed, we pass around the final `end` time rather than a relative sleep time.